### PR TITLE
refactor: single code base for both text and drawables & other bug fixes

### LINKED ISSUE #189: Single Code for Text and Bitmaps
**Describe the Bug**
Currently, Sending the bytes to the device is different for Text and Drawable. I would suggest to create a common code base for the above.

**Additional context**
The code for text has to be altered completely but I can assure creating a single class that can handle both text and drawable hexadecimals.

So, I will first try to convert all the text into hexadecimal strings and then send it to the function.

**Would you like to work on the issue?**
Yes Sir. But I cannot confirm the time required, according to me, this would be completed before this weekend

@iamareebjamal, please let me know your thoughts on this sir and if I should proceed with it

### PR DESCRIPTION
**Pull request by adityastic**
Fixes #189 

Changes: 
- Optimized Code
- Concept of Text is completely removed, we convert everything to the hexadecimal strings and then convert it to byte array
- Renamed BitmapDatatoSend to DataToSend because it is a single class now.
- Fixed a Bug which did not let you send the drawable to device because of the `content.text.isEmpty()`. Now it checks the radio button and then sends the bytearray
- Removed unused resources from the app
- Removed Kotlin Unit Tests as they depended on the Text Logic
- Lint Errors Fixed

### CODE REVIEW

**Comment by adityastic** [comment]
@iamareebjamal fixed almost all lint errors, excluding the missing icon. Please have a look

**Comment by iamareebjamal** [comment]
Peer Review First

**Comment by adityastic** [comment]
> Peer Review First

On it Sir.@yashk2000 @yatri1609 @anhanh11001 Please Peer Review

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -20,14 +20,14 @@
...
     <string name="permission_required">Permission required</string>
-    <string name="txt_turn_on_bluetooth">Turn on Bluetooth and try again</string>
     <string name="no_device_found">No device found. Please try again.</string>
     <string name="select_drawable_text">Select Drawable:</string>
-    <string name="select_drawable">Please select Drawable to be previewed</string>
     <string name="character_not_found">Some of the character(s) entered cannot be displayed.</string>
     <string name="send_text">Send Text</string>
     <string name="send_drawables">Send Drawables</string>
     <string name="invertLED">Invert LED</string>
+    <string name="radio_error">Select option before proceeding</string>
```
Comment (yashk2000) [inline app/src/main/res/values/strings.xml]: Where is this used?

Comment (adityastic) [reply app/src/main/res/values/strings.xml]: Ah I removed the use. Thanks sir, just a min

Comment (adityastic) [reply app/src/main/res/values/strings.xml]: Have loik

Comment (adityastic) [reply app/src/main/res/values/strings.xml]: Fixed, have a look sir

**Comment by adityastic** [comment]
@yashk2000 Thanks. @iamareebjamal Peer review done

File: app/build.gradle [hunk 2]
```diff
@@ -38,6 +38,8 @@ spotless {
 }
 
 dependencies {
+    // Kotlin Libraries
+    implementation "org.jetbrains.kotlin:kotlin-reflect:$kotlin_version"
```
Comment (iamareebjamal) [inline app/build.gradle]: Why is this needed. It is 2 MB

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -14,13 +14,16 @@
     <application
         android:name=".ui.App"
         android:allowBackup="true"
+        android:fullBackupContent="true"
```
Comment (iamareebjamal) [inline app/src/main/AndroidManifest.xml]: Why

File: app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt [hunk 4]
```diff
@@ -328,13 +310,39 @@ class MessageActivity : AppCompatActivity() {
...
+                                        else if (!invertLED.isChecked) " "
+                                        else "",
+                                        invertLED.isChecked
+                                ).second,
+                                flash.isChecked, marquee.isChecked,
+                                Speed.values()[speed.selectedItemPosition],
+                                Mode.values()[mode.selectedItemPosition]
+                        )
+                )
+        )
```
Comment (iamareebjamal) [inline app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]: Too much indentation

File: app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt [hunk 5]
```diff
@@ -328,13 +310,39 @@ class MessageActivity : AppCompatActivity() {
...
+                                        drawableRecyclerAdapter.getSelectedItem()?.image
+                                                ?: resources.getDrawable(R.drawable.apple),
+                                        invertLED.isChecked),
+                                flash.isChecked,
+                                marquee.isChecked,
+                                Speed.values()[speed.selectedItemPosition],
+                                Mode.values()[mode.selectedItemPosition]
+                        )
+                )
+        )
```
Comment (iamareebjamal) [inline app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]: Same

File: app/src/main/java/com/nilhcem/blenamebadge/util/Converters.kt [hunk 6]
```diff
@@ -137,4 +138,22 @@ object Converters {
...
+        var valid = true
+        val list = mutableListOf<String>()
+        for (letter in data) {
+            if (DataToByteArrayConverter.CHAR_CODES.containsKey(letter)) {
+                list.add(
+                        if (invertLED)
+                            invertHex(DataToByteArrayConverter.CHAR_CODES.getValue(letter))
+                        else
+                            DataToByteArrayConverter.CHAR_CODES.getValue(letter)
+                )
```
Comment (iamareebjamal) [inline app/src/main/java/com/nilhcem/blenamebadge/util/Converters.kt]: Too much indentation

File: app/build.gradle (hunk 2 above)
Comment (iamareebjamal) [reply app/build.gradle]: First of all, I see no use of this library, and even if it was used, I would have voted against it

Comment (adityastic) [reply app/build.gradle]: Alright sir. I added this because lint told me too

Comment (adityastic) [reply app/build.gradle]: Let me remove it

File: app/src/main/AndroidManifest.xml (hunk 3 above)
Comment (adityastic) [reply app/src/main/AndroidManifest.xml]: using fullBackupContent backs up the user data when a person uninstalls the application, he can restore the state after reinstalling the application

Comment (iamareebjamal) [reply app/src/main/AndroidManifest.xml]: How is it related to this issue?

File: app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt (hunk 4 above)
Comment (adityastic) [reply app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]: Ok sir, let me make it a bit compact

File: app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt (hunk 5 above)
Comment (adityastic) [reply app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]: On it

File: app/src/main/java/com/nilhcem/blenamebadge/util/Converters.kt (hunk 6 above)
Comment (adityastic) [reply app/src/main/java/com/nilhcem/blenamebadge/util/Converters.kt]: On it sir

File: app/src/main/AndroidManifest.xml (hunk 3 above)
Comment (adityastic) [reply app/src/main/AndroidManifest.xml]: It was one of the lint fixes suggested @iamareebjamal.

**Comment by adityastic** [comment]
@iamareebjamal Thank you sir, I found 8 tab indentation issues in many files. I've changed all of them

**Comment by adityastic** [comment]
@iamareebjamal have a look at the fixes sir

File: app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt [hunk 7]
```diff
@@ -331,13 +310,39 @@ class MessageActivity : AppCompatActivity() {
         builder.create().show()
     }
 
-    private fun convertToDeviceDataModel(): DataToSend {
-        return DataToSend(listOf(Message(content.text.trim().toString(), flash.isChecked, marquee.isChecked, Speed.values()[speed.selectedItemPosition], Mode.values()[mode.selectedItemPosition])), invertLED.isChecked)
+    private fun convertTextToDeviceDataModel(): DataToSend {
+        return DataToSend(
+            listOf(
```
Comment (iamareebjamal) [inline app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]: Move to previous line

File: app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt [hunk 8]
```diff
@@ -331,13 +310,39 @@ class MessageActivity : AppCompatActivity() {
...
+                Message(
```
Comment (iamareebjamal) [inline app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]: Move to previous line

File: app/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -20,14 +20,13 @@
     <string name="mode_animation">Animation</string>
     <string name="mode_laser">Laser</string>
     <string name="send_button">SEND</string>
-    <string name="preview_button">PREVIEW</string>
     <string name="permission_required">Permission required</string>
-    <string name="txt_turn_on_bluetooth">Turn on Bluetooth and try again</string>
     <string name="no_device_found">No device found. Please try again.</string>
     <string name="select_drawable_text">Select Drawable:</string>
-    <string name="select_drawable">Please select Drawable to be previewed</string>
```
Comment (naman653) [inline app/src/main/res/values/strings.xml]: Please don't remove these strings for now. Might be useful.

File: app/src/main/res/values/strings.xml [hunk 10]
```diff
@@ -20,14 +20,13 @@
     <string name="mode_animation">Animation</string>
     <string name="mode_laser">Laser</string>
     <string name="send_button">SEND</string>
-    <string name="preview_button">PREVIEW</string>
     <string name="permission_required">Permission required</string>
-    <string name="txt_turn_on_bluetooth">Turn on Bluetooth and try again</string>
```
Comment (naman653) [inline app/src/main/res/values/strings.xml]: Same

File: app/src/main/res/values/strings.xml (hunk 9 above)
Comment (adityastic) [reply app/src/main/res/values/strings.xml]: Sir, we are using automatic preview now, the second one will definitely not be used. We can keep the first one though

File: app/src/main/res/values/strings.xml (hunk 10 above)
Comment (adityastic) [reply app/src/main/res/values/strings.xml]: Yes Sir, sure. let me undo that

File: app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt (hunk 8 above)
Comment (adityastic) [reply app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]: ok sir

File: app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt (hunk 7 above)
Comment (adityastic) [reply app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]: yes sir, I am learning the type of indentation required now. just a min

Comment (adityastic) [reply app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]: made the changes, have a look

**Comment by adityastic** [comment]
@iamareebjamal @naman653 Done, check the new changes. let me know if anything else is to be fixed

**Comment by adityastic** [comment]
@iamareebjamal when I was going through the code right now. I also found that KeyboardUtils created is a duplicate. We already have a view extension class called 
[https://github.com/fossasia/badge-magic-android/blob/development/app/src/main/java/com/nilhcem/blenamebadge/core/android/ext/ViewExt.kt](url)
 that does this for us. So, I removed it in this update and fixed the working logic

File: app/src/main/res/values/strings.xml (hunk 10 above)
Comment (yashk2000) [reply app/src/main/res/values/strings.xml]: Is this resource really needed @naman653? We are using `enable_bluetooth` to warn the user when bluetooth is off. This string is redundant in any case according to me.

Comment (adityastic) [reply app/src/main/res/values/strings.xml]: @naman653 Yeah TBH @yashk2000 is right
